tensorflow/compiler/xla/service/gpu/gemm_thunk.cc

```diff
@@ -293,80 +398,148 @@ Status RunGemm(const GpuGemmConfig &gemm_config,
+  std::tie(lhs_matrix, rhs_matrix, output_matrix) = PopulateInputOutputMatrices(
+      gemm_config, lhs_buffer, rhs_buffer, output_buffer);
+  const Shape &output_shape = gemm_config.output_shape;
+  int64_t batch_size = gemm_config.backend_config.batch_size();
+  const GemmBackendConfig &backend_config = gemm_config.backend_config;
+  // The BlasLtMatmul routines (unlike BlasGemm, BlasGemmBatched etc.) take
+  // alpha and beta with the same type as the matrices.
   complex128 alpha = {backend_config.alpha_real(), backend_config.alpha_imag()};
   double beta = backend_config.beta();
 
-  switch (output_shape.element_type()) {
-    case S32: {
-      if (!best_algorithm) {
-        return InternalError("Only extended GEMM is supported for int32_t");
+  // The BlasLtMatmul routines are only supported from CUDA 11.0 onward.
+  if (gemm_config.use_cublaslt && stream->parent()->SupportsBlasPlans() &&
+      BlasPlansCompatibleType(output_shape.element_type())) {
+    se::blas::IBlasLtMatmulAlgorithm *best_algorithm = nullptr;
+    if (algorithm_being_profiled) {
+      best_algorithm = algorithm_being_profiled;
+    }
+    complex128 beta_cmplx = {beta, 0};
+    switch (output_shape.element_type()) {
+      case F16:
+        CHECK_EQ(alpha.imag(), 0);
+        return DoGemmLt<Eigen::half>(
+            batch_size, lhs_matrix, rhs_matrix, output_matrix, stream,
+            static_cast<Eigen::half>(alpha.real()),
+            static_cast<Eigen::half>(beta), scratch_allocator, best_algorithm,
+            /*output_profile_result=*/profile_result);
+      case F32:
+        CHECK_EQ(alpha.imag(), 0);
+        return DoGemmLt<float>(
+            batch_size, lhs_matrix, rhs_matrix, output_matrix, stream,
+            static_cast<float>(alpha.real()), static_cast<float>(beta),
+            scratch_allocator, best_algorithm,
+            /*output_profile_result=*/profile_result);
+      case F64:
+        CHECK_EQ(alpha.imag(), 0);
+        return DoGemmLt<double>(batch_size, lhs_matrix, rhs_matrix,
+                                output_matrix, stream,
+                                static_cast<double>(alpha.real()), beta,
+                                scratch_allocator, best_algorithm,
+                                /*output_profile_result=*/profile_result);
+      case C64:
+        return DoGemmLt<complex64>(
+            batch_size, lhs_matrix, rhs_matrix, output_matrix, stream,
+            static_cast<complex64>(alpha), static_cast<complex64>(beta_cmplx),
+            scratch_allocator, best_algorithm,
+            /*output_profile_result=*/profile_result);
+      case C128:
+        return DoGemmLt<complex128>(batch_size, lhs_matrix, rhs_matrix,
+                                    output_matrix, stream, alpha, beta_cmplx,
+                                    scratch_allocator, best_algorithm,
+                                    /*output_profile_result=*/profile_result);
+    }
```

Comment:
@philipphack What ought to be done for the remaining cases/types in the switch here?

Reply:
Other types currently aren't supported by cuBLASLt, and `BlasPlansCompatibleType` returns false in these cases.

Reply:
That's what I figured. Linters still complain in this case, though.

Reply:
I added a default case that triggers an error.